access_control.py

```diff
@@ -0,0 +1,225 @@
+        with open(file_path, "r") as f:
+            feedstocks = f.readlines()
+            feedstocks = [feedstock.strip() for feedstock in feedstocks]
+            file_name = file_path.parts[-1].strip(".txt")
+            file_name_feedstock_mapping[file_name] = feedstocks
+    return file_name_feedstock_mapping
+
+
+def _process_access_control_requests(path, remove=True):
+    print(f"Processing access control requests for {path}")
+    grant_access_request = _get_filename_feedstock_mapping(path)
+    for filename, feedstocks in grant_access_request.items():
+        if filename in CIRUN_FILENAME_RESOURCE_MAPPING:
+            resource_mapping = CIRUN_FILENAME_RESOURCE_MAPPING.get(filename)
+            resource = resource_mapping.get("resource")
+            policy_args = resource_mapping.get("policy_args")
+            for feedstock in feedstocks:
+                feedstock_repo = f"{feedstock}-feedstock"
+                print(f"Processing feedstock for access control: {feedstock_repo}")
+                _process_request_for_feedstock(
+                    feedstock_repo, resource, remove=remove, policy_args=policy_args
+                )
+                action = "remove" if remove else "add"
+                update_access_yaml(resource, feedstock_repo, action=action)
+
+
+def process_access_control_requests():
+    """Process access control requests"""
+    print("Processing access control request")
+    _process_access_control_requests("grant_access", remove=False)
+    _process_access_control_requests("revoke_access")
+
+
+def _process_request_for_feedstock(feedstock, resource, remove, policy_args):
+    feedstock_clone_path = tempfile.mkdtemp()
+    assert GH_ORG
+    clone_cmd = f"git clone --depth 1 https://github.com/{GH_ORG}/{feedstock}.git {feedstock_clone_path}"
+    print(f"Cloning: {clone_cmd}")
+    subprocess.run(
+        clone_cmd,
+        shell=True,
+    )
+
+    register_ci_cmd = [
+        "conda-smithy register-ci",
+        f"--organization {GH_ORG}",
+        "--without-azure",
+        "--without-travis",
+        "--without-circle",
+        "--without-appveyor",
+        "--without-drone",
+        "--without-webservice",
+        "--without-anaconda-token",
+        f"--feedstock_directory {feedstock_clone_path}",
+        f"--cirun-resources {resource}",
+    ]
+
+    if policy_args:
+        policy_args_param = [f"--cirun-policy-args {arg}" for arg in policy_args]
+        register_ci_cmd.extend(policy_args_param)
+    if remove:
+        register_ci_cmd.append("--remove")
+
+    register_ci_cmd_str = " ".join(register_ci_cmd)
+    print(f"RegisterCI command: {register_ci_cmd_str}")
+    subprocess.check_call(register_ci_cmd_str,
+        shell=True,
+    )
+
+
+def check_if_repo_exists(feedstock_name):
+    repo = f"{feedstock_name}-feedstock"
+    owner_repo = f"{GH_ORG}/{repo}"
+    print(f"Checking if {owner_repo} exists")
+    response = requests.get(f"https://api.github.com/repos/{owner_repo}")
```

Comment:
To save API calls (I know 😂) we usually just query the regular website, as in:

```suggestion
    response = requests.get(f"https://github.com/{GH_ORG}")
```